AppinfoJsonFile.js

```diff
@@ -210,6 +211,40 @@ AppinfoJsonFile.prototype.getTranslationSet = function() {
 // we don't localize or write appinfo.json source files
 AppinfoJsonFile.prototype.write = function() {};
 
+AppinfoJsonFile.prototype.formatPath = function (template, parameters) {
+    var pathname = parameters.sourcepath || "";
+    var localepath = parameters.localepath;
+    var output = "";
+    var resourceDir = this.project.getResourceDirs("json")[0] || ".";
+
+    for (var i = 0; i < template.length; i++) {
+        if ( template[i] !== '[' ) {
+            output += template[i];
+        } else {
+            var start = ++i;
+            while (i < template.length && template[i] !== ']') {
+                i++;
+            }
+            var keyword = template.substring(start, i);
+            switch (keyword) {
+                case 'dir':
+                    output += path.dirname(pathname);
+                    break;
+                default:
+                case 'filename':
+                    output += path.basename(pathname);
+                    break;
+                case 'resDir':
+                    output += resourceDir;
+                    break;
+                case 'locale':
+                    output +=localepath;
+                    break;
+            }
+        }
+    }
+    return path.normalize(output);
```

Comment:
You can use this.API.utils.formatPath() instead of these lines

Reply:
there's reason that I didn't use loctool utils formatPath.

   1. locale directory. I don't want to write  every locale list to to mapping. mostrly I can get it from ilib in code.
   2. resDir option. We could modify the output directory name which we get from configfile..
   `"resourceDirs": { "json":"resources"  },`  /    `"resourceDirs": { "json":"res"  },`
    one app can have /resources/ko/string.json
    some app want to have ./res/ko/string.json

Reply:
updated to use loctool's formatPath(). also updated loctool code as well.
https://github.com/iLib-js/loctool/pull/223